src/test/java/com/fasterxml/jackson/core/JsonGeneratorTest.java

```diff
@@ -0,0 +1,207 @@
+package com.fasterxml.jackson.core;
+
+import com.fasterxml.jackson.core.filter.FilteringGeneratorDelegate;
+import com.fasterxml.jackson.core.filter.TokenFilter;
+import com.fasterxml.jackson.core.io.IOContext;
+import com.fasterxml.jackson.core.json.ReaderBasedJsonParser;
+import com.fasterxml.jackson.core.json.UTF8JsonGenerator;
+import com.fasterxml.jackson.core.json.WriterBasedJsonGenerator;
+import com.fasterxml.jackson.core.sym.CharsToNameCanonicalizer;
+import com.fasterxml.jackson.core.type.WritableTypeId;
+import com.fasterxml.jackson.core.util.BufferRecycler;
+import com.fasterxml.jackson.core.util.ByteArrayBuilder;
+import com.fasterxml.jackson.core.util.JsonGeneratorDelegate;
+
+import java.io.CharArrayWriter;
+import java.io.IOException;
+import java.io.PrintStream;
+
+/**
+ * Unit tests for class {@link JsonGenerator}.
+ *
+ * @date 2017-09-19
+ * @see JsonGenerator
+ **/
+public class JsonGeneratorTest extends BaseTest {
+    public void testWriteObjectThrowsIllegalStateException() throws IOException {
+        BufferRecycler bufferRecycler = new BufferRecycler();
```

Comment:
In general I much prefer going through public api, constructing generators and parsers using `JsonFactory`.
Only if that is not possible for some reason it should be necessary to use internal api.
One reason is that internal api will keep on changing and tests that rely on them are quite fragile, needing to be updated if and when call sequences change.